engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/ExecutableMetaData.java

```diff
@@ -122,15 +124,6 @@ private ExecutableMetaData(
 	 * @return Meta data for the specified parameter. Will never be {@code null}.
 	 */
 	public ParameterMetaData getParameterMetaData(int parameterIndex) {
-		if ( parameterIndex < 0 || parameterIndex > parameterMetaDataList.size() - 1 ) {
-			throw log.getInvalidExecutableParameterIndexException(
-					ExecutableElement.getExecutableAsString(
-							getType().toString() + "#" + getName(),
-							parameterTypes
-					), parameterTypes.length
-			);
-		}
-
```

Comment:
Yes, it wasn't needed really. The only invocation path not originating from tests already makes sure that the given parameter values match the length of the validated executable (see `ValidatorImpl#validateParametersForGroup()`). Hence this check never triggered anyways.